(DOCSP-19881): Explicitly state that project is inclusive only (#1901)
mongodb/docs-realm: source/triggers/database-triggers.txt

limit the number of fields that the Trigger processes by using a 
:manual:`$project </reference/operator/aggregation/project>` expression. 

> .. note:: Project is inclusive only

Reviewer: worth mentioning that there's an exception for `_id`, esp b/c this is in the example below.
<img width="1076" alt="Screen Shot 2022-05-04 at 2 39 40 PM" src="https://user-images.githubusercontent.com/90647379/166803733-4f8bf959-addf-4ebc-807c-57fe2803e52e.png">

Author: Ahh, interesting. I admit I did not read the linked HELP ticket. After going over it, I've updated the note and I _think_ it now includes all the relevant details.

>   A projection expression is inclusive. If you want to exclude a single 

Reviewer: i think it could be worthwhile adding a sentence expanding on what the implication of projections being "inclusive only" means. something like:

```
You can only specify the fields that you want to include in your project expression, not those you want to exclude. 
```

Author: I had a sentence like that in my first pass at this, but I found it confusing and didn't feel it added helpful info. I did update this note based on the link HELP, so hopefully it's clearer now and no longer needs this?

  field, the projection expression must include every field *except* the 

>   one you want to exclude.

Reviewer: might also be worth including a teeny example. something like:

```
{
   "my_field": <1>
   
}
```

> 

Reviewer: also since we're just talking about the MongoDB aggregation behavior, rather than anything realm-specific, might be worth adding a line at bottom of the note like, "For further information on the features and limitations of projections, refer to the [MongoDB $project documentation](https://www.mongodb.com/docs/manual/reference/operator/aggregation/project/)"

Author: Before reading the linked HELP, I was prepared to add this, but after looking at those messages, I think the `operation_type` requirement and the fact that project expressions can _only_ be inclusive in a Trigger context make it different enough that linking them over to the Project docs might just add confusion.

limit the number of fields that the Trigger processes by using a 
:manual:`$project </reference/operator/aggregation/project>` expression. 

.. note:: Project is inclusive only

>   When using Triggers, a projection expression is inclusive *only*. 
>   Project does not support mixing inclusions and exclusions. When using 
>   Triggers, you must include ``operation_type``, which means the project 
>   expression must be an inclusive one.

Reviewer: ```suggestion
  When using Triggers, a projection expression is inclusive *only*. 
  Project does not support mixing inclusions and exclusions.
  The project expression must be inclusive because Triggers require you 
  to include ``operation_type``. 
```

Reviewer: changed order of phrases for clarity

  
  If you want to exclude a single field, the projection expression must 

>   include every field *except* the one you want to exclude. The exception 
>   is ``_id``, which is included by default.

Reviewer: ```suggestion
  include every field *except* the one you want to exclude.
  You can only explicitly exclude ``_id``, which is included by default.
```
